File: webapp-net/DynamicDocumentation/Models/Topic.cs
using System;
using Newtonsoft.Json;
using Sdl.Web.Common.Models;

namespace Sdl.Web.Modules.DynamicDocumentation.Models
{
    /// <summary>
    /// Topic Entity
    /// </summary>
    [SemanticEntity(Vocab = SchemaOrgVocabulary, EntityName = "Topic", Prefix = "s", Public = true)]
    [Serializable]
    public class Topic : EntityModel
    {
        [SemanticProperty("topicBody")]
        [JsonProperty(PropertyName = "topicBody")]
        public RichText TopicBody { get; set; }

        [SemanticProperty("topicTitle")]
        [JsonProperty(PropertyName = "topicTitle")]
        public string TopicTitle { get; set; }
    }
}

File: webapp-java/dxa-module-dynamic-documentation/src/main/java/com/sdl/dxa/modules/ish/model/Topic.java
package com.sdl.dxa.modules.ish.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.sdl.webapp.common.api.mapping.semantic.annotations.SemanticEntity;
import com.sdl.webapp.common.api.mapping.semantic.annotations.SemanticProperty;
import com.sdl.webapp.common.api.model.RichText;
import com.sdl.webapp.common.api.model.entity.AbstractEntityModel;
import lombok.Data;
import lombok.EqualsAndHashCode;

import static com.sdl.webapp.common.api.mapping.semantic.config.SemanticVocabulary.SCHEMA_ORG;

/**
 * Topic model.
 */
@SemanticEntity(entityName = "Topic", vocabulary = SCHEMA_ORG, prefix = "s", public_ = true)
@Data
@EqualsAndHashCode(callSuper = true)
public class Topic extends AbstractEntityModel {

    @SemanticProperty("topicBody")
    @JsonProperty("topicBody")
    private RichText topicBody;

    @SemanticProperty("topicTitle")
    @JsonProperty("topicTitle")
    private String topicTitle;

    public RichText getTopicBody() {
        return topicBody;
    }

    public void setTopicBody(RichText topicBody) {
        this.topicBody = topicBody;
    }

    public String getTopicTitle() {
        return topicTitle;
    }

    public void setTopicTitle(String topicTitle) {
        this.topicTitle = topicTitle;
    }
}
